src/screens/IdentityManagement.js

```diff
@@ -39,17 +42,44 @@ function IdentityManagement({ accounts, navigation }) {
 	const { currentIdentity } = accounts.state;
 	if (!currentIdentity) return null;
 
-	const onOptionSelect = value => {
+	const onDelete = async () => {
+		const deleteSucceed = await accounts.deleteCurrentIdentity();
+		if (deleteSucceed) {
+			navigateToLandingPage(navigation, true);
+		} else {
+			alertIdentityDeletionError();
+		}
+	};
+
+	const toggleBiometric = async () => {
+		if (currentIdentity.biometricEnabled) {
+			await unlockIdentitySeedWithBiometric(currentIdentity);
+			await accounts.identityDisableBiometric();
+		} else {
+			const pin = await unlockPin(navigation);
+			navigation.pop();
+			await accounts.identityEnableBiometric(pin).catch(error => {
+				// error here is likely no fingerprints/biometrics enrolled, so should be displayed to the user
+				alertBiometricError(error);
+			});
+		}
+	};
+
+	const onOptionSelect = async value => {
 		if (value === 'PathDelete') {
 			alertDeleteIdentity(async () => {
-				await unlockSeedPhrase(navigation);
-				const deleteSucceed = await accounts.deleteCurrentIdentity();
-				if (deleteSucceed) {
-					navigateToLandingPage(navigation, true);
+				if (currentIdentity.biometricEnabled) {
+					await unlockIdentitySeedWithBiometric(currentIdentity)
```

Comment:
`await` and `promise.then()` do not occurs at same time, please use `await` instead